test/clj/chronograph/handlers/organization_test.clj

```diff
@@ -0,0 +1,75 @@
+(ns chronograph.handlers.organization-test
+  (:require [chronograph.domain.acl :as acl]
+            [chronograph.factories :as factories]
+            [chronograph.fixtures :as fixtures]
+            [chronograph.handlers.organization :as organization]
+            [clojure.spec.alpha :as s]
+            [clojure.test :refer :all])
+  (:import org.postgresql.util.PSQLException))
+
+(use-fixtures :once fixtures/config fixtures/datasource)
+(use-fixtures :each fixtures/clear-db)
+
+(deftest create-new-organization-first-time
+  (testing "Creating a new organization returns a valid organization map in the response, and the creator is registered as admin in the ACL."
+    (let [user (factories/create-user)
+          response (organization/create {:body {:name "foo" :slug "bar"}
+                                         :user user})]
+      (is (= 200 (:status response)))
+      (is (s/valid? :organizations/organization
+                    (:body response)))
+      (is (acl/admin? (:users/id user)
+                      (:organizations/id (:body response)))))))
+
+
+(deftest create-organization-when-name-exists
+  (testing "Creating an organization with pre-existing name works if slugs are different."
+    (let [user (factories/create-user)]
+      (is (s/valid? :organizations/organization
+                    (:body (do (organization/create {:body {:name "foo" :slug "bar"}
+                                                     :user user})
+                               (organization/create {:body {:name "foo" :slug "bar-baz"}
+                                                     :user user}))))))))
+
+
+(deftest create-organization-when-slug-exists
```

Comment:
This is already tested in the domain layer, I don't think we need to repeat it here.

For a handler level test, it would make more sense to check the HTTP response codes. For example, maybe the handler should return a 400 if the slug already exists.

Reply:
Fixed. This now tests for HTTP 500, which is how our handler currently behaves.